### PR TITLE
valentina: downgrade qt to 5.12

### PR DESCRIPTION
**Pull request by jnetod**
###### Motivation for this change

ZHF: #122042

Support for late Qt versions doesn't landed in a stable release yet:

- https://gitlab.com/smart-pattern/valentina/-/commit/3b728f10dae114ea76f97b5e75a775a9364bfb52
- https://gitlab.com/smart-pattern/valentina/-/commit/6864ddc1af63074b811a7c54fb9d37d0b141fa02
- https://gitlab.com/smart-pattern/valentina/-/commit/6af9d104cdc9c9a2a008a0b5fca63d907c16a8f1 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
cc @NixOS/qt-kde 

I would like to avoid having people have multiple versions of qt on their system if possible.

**Comment by samueldr** [comment]
Given this seems to be "*specialist*" software, it's probably much less of an issue than if it was a popular often-used tool.

**Comment by doronbehar** [comment]
> Support for late Qt versions doesn't landed in a stable release yet:
> 
>     * [gitlab.com/smart-pattern/valentina/-/commit/3b728f10dae114ea76f97b5e75a775a9364bfb52](https://gitlab.com/smart-pattern/valentina/-/commit/3b728f10dae114ea76f97b5e75a775a9364bfb52)
> 
>     * [gitlab.com/smart-pattern/valentina/-/commit/6864ddc1af63074b811a7c54fb9d37d0b141fa02](https://gitlab.com/smart-pattern/valentina/-/commit/6864ddc1af63074b811a7c54fb9d37d0b141fa02)
> 
>     * [gitlab.com/smart-pattern/valentina/-/commit/6af9d104cdc9c9a2a008a0b5fca63d907c16a8f1](https://gitlab.com/smart-pattern/valentina/-/commit/6af9d104cdc9c9a2a008a0b5fca63d907c16a8f1)
> 
> 

How about adding these patches to the package and make it use `libsForQt5`?

**Comment by jnetod** [comment]
@doronbehar I've tried to cherry-pick those commits, no luck; too much changes in between, those commits alone doesn't help.

**Comment by doronbehar** [comment]
> @doronbehar I've tried to cherry-pick those commits, no luck; too much changes in between, those commits alone doesn't help.

The upstream commits you referenced should be merged into their latest release - For instance https://gitlab.com/smart-pattern/valentina/-/blob/v0.7.46/src/libs/vlayout/vlayoutdef.h includes the `QPainterPath` is included there. I'm rather positive you should be able to update the package with qt5.15, the current version in nixpkgs is 0.6.1 from 2 years ago.

**Comment by jnetod** [comment]
v0.7.x is a test build. Should we maintain both versions, newer only, or stable only?

> Currently, we have two different versions: stable and test builds. I propose to change the schema. Each version with even number like 0.6.0, 0.8.0, 1.0.0 and so on will be stable release. Each with odd number 0.7.0, 0.9.0, 1.1.0 will be a version for testing. Instead of publishing a test build each time we will publish a new release for the test version. At the end of the test cycle, a new stable version will be released. For example, 0.7.0 will be published as 0.8.0. Testing will be continued in 0.9.0.

https://gitlab.com/smart-pattern/valentina/-/issues/5

**Comment by doronbehar** [comment]
> v0.7.x is for test builds. Should we maintain both versions, newer only or stable only?

:+1: for noticing this. That's a good question and I don't have a strong opinion, there's not even a strong consensus among other repos https://repology.org/project/valentina/versions .

cc @jfrankenau maintainer for an opinion. I think that if the update doesn't introduce too many changes we should update it to what upstream considers as unstable yet.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26582,7 +26582,7 @@ in
 
   utox = callPackage ../applications/networking/instant-messengers/utox { };
 
-  valentina = libsForQt514.callPackage ../applications/misc/valentina { };
+  valentina = libsForQt512.callPackage ../applications/misc/valentina { };
```
Comment (doronbehar) [inline pkgs/top-level/all-packages.nix:26585]: In anycase, if the PR will be merged as is, please add:

```suggestion
  # Version 0.8.x or above should support qt5.15
  valentina = libsForQt512.callPackage ../applications/misc/valentina { };
```

**Comment by jonringer** [comment]
If the update works with little issues, I would prefer that route.

Unfortunate that upstream is significantly far behind qt development.

**Review by jonringer — APPROVED** [review]
I would rather have a working application with closure bloat, than a failing one..

```
https://github.com/NixOS/nixpkgs/pull/122211

1 package built:
valentina
```

Hopefully we can update this to 515 once upstream supports it.

**Pull request merged**